src/clients/github.js

```diff
@@ -0,0 +1,276 @@
+import get from 'lodash/get';
+import GitHub from 'github-api';
+import isEmpty from 'lodash/isEmpty';
+import trim from 'lodash/trim';
+import performWithRetries from '../util/performWithRetries';
+import generatePreview from '../util/generatePreview';
+
+const _anonymousGithub = new GitHub({});
+const COMMIT_MESSAGE = 'Created using Popcode: https://popcode.org';
+const MASTER = 'master';
+const GH_PAGES = 'gh-pages';
+const NETWORK_ERROR = 'Network Error';
+const parser = new DOMParser();
+
+export function EmptyGistError(message) {
+  this.name = 'EmptyGistError';
+  this.message = message;
+  this.stack = new Error().stack;
+}
+EmptyGistError.prototype = Object.create(Error.prototype);
+
+function getPageTitle(preview) {
+  const parsedDocument = parser.parseFromString(preview, 'text/html');
+
+  const titleWithoutPunctuationAndWhitespace = parsedDocument.title.
+    replace(/[^\w\s]|_/g, '').
+    replace(/\W/g, '-');
+
+  return titleWithoutPunctuationAndWhitespace;
+}
+
+export async function createRepoFromProject(project, user) {
+  const github = clientForUser(user);
+  const preview = generatePreview(project);
+  const title = getPageTitle(preview);
+
+  const REPO_NAME = `${title}-${Date.now()}`;
+
+  const {data} =
+    await performWithRetryNetworkErrors(
+      () => github.getUser().createRepo({name: REPO_NAME, auto_init: true}),
```

Comment:
yeah, according to [this Stack Overflow thread](https://stackoverflow.com/questions/18565425/how-do-i-create-an-orphan-branch-from-the-github-api) it isn’t possible to create an orphan branch via the GitHub API. obviously it’s not ideal to have the two commits adding and removing the README in the history, but i’m much less concerned about that than about having the README actually present in the repo’s files. so I think we’re good here : )